### ISSUE TITLE
Kopia volume backup is expired, gets deleted but not removed

### ISSUE THREAD

**Issue by notz** [vmware-tanzu/velero#5824]
**What steps did you take and what happened:**
I have some kopia volume backups. First the deletion was working fine, but know i have some problems.
Velero detects that an old backup is expired and starts und succeeds the deletion request:

```
time="2023-02-02T05:58:37Z" level=info msg="The request has been processed, skip." controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-qbl89 logSource="pkg/controller/backup_deletion_controller.go:137" |  
time="2023-02-02T05:58:37Z" level=info msg="Reconciliation done" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-qbl89 logSource="pkg/controller/backup_deletion_controller.go:390" |  
time="2023-02-02T05:58:37Z" level=info msg="Removing restores" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-qbl89 logSource="pkg/controller/backup_deletion_controller.go:324" |  
time="2023-02-02T05:58:36Z" level=info msg="Removing backup from backup storage" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-qbl89 logSource="pkg/controller/backup_deletion_controller.go:318" |  
time="2023-02-02T05:58:36Z" level=info msg="Removing pod volume snapshots" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-qbl89 logSource="pkg/controller/backup_deletion_controller.go:310" |  
time="2023-02-02T05:58:36Z" level=info msg="Removing PV snapshots" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-qbl89 logSource="pkg/controller/backup_deletion_controller.go:285" |  
time="2023-02-02T05:58:36Z" level=info msg="Removing existing deletion requests for backup" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-qbl89 logSource="pkg/controller/backup_deletion_controller.go:427" |  
time="2023-02-02T05:58:33Z" level=info msg="Creating a new deletion request" backup=backup/plausible-clickhouse-lt-20230126053553 expiration="2023-02-02 05:35:53 +0000 UTC" logSource="pkg/controller/gc_controller.go:186" |  
time="2023-02-02T05:58:33Z" level=info msg="Backup:plausible-clickhouse-lt-20230126053553 has expired" backup=backup/plausible-clickhouse-lt-20230126053553 expiration="2023-02-02 05:35:53 +0000 UTC" logSource="pkg/controller/gc_controller.go:126"
```

But the backup is still there:
```
Name:         plausible-clickhouse-lt-20230126053553
Namespace:    backup
Labels:       kustomize.toolkit.fluxcd.io/name=service
              kustomize.toolkit.fluxcd.io/namespace=flux-system
              velero.io/schedule-name=plausible-clickhouse-lt
              velero.io/storage-location=default
Annotations:  velero.io/source-cluster-k8s-gitversion=v1.24.8
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=24

Phase:  Deleting

Errors:    0
Warnings:  0

Namespaces:
  Included:  plausible
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        storageclasses.storage.k8s.io
  Cluster-scoped:  auto

Label selector:  app.kubernetes.io/instance=clickhouse

Storage Location:  default

Velero-Native Snapshot PVs:  auto

TTL:  168h0m0s

CSISnapshotTimeout:  10m0s

Hooks:  <none>

Backup Format Version:  1.1.0

Started:    2023-01-26 06:35:53 +0100 CET
Completed:  2023-01-26 06:36:21 +0100 CET

Expiration:  2023-02-02 06:35:53 +0100 CET

Total items to be backed up:  10
Items backed up:              10

Velero-Native Snapshots: <none included>

Deletion Attempts (1 failed):
  2023-02-03 08:58:33 +0100 CET: Processed
  Errors:
    error to open backup repo: repo config /home/nonroot/udmrepo/repo-5263a16b-e3a5-41aa-a51b-f6b318a49255.conf doesn't exist: stat /home/nonroot/udmrepo/repo-5263a16b-e3a5-41aa-a51b-f6b318a49255.conf: no such file or directory

CSI Volume Snapshots: <none included>

kopia Backups (specify --details for more information):
  Completed:  1
```

Subsequent deletion requests are failing, because the backup files are already removed from store:
```
time="2023-02-02T06:58:35Z" level=info msg="The request has been processed, skip." controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-k6zwz logSource="pkg/controller/backup_deletion_controller.go:137" |  
time="2023-02-02T06:58:33Z" level=info msg="Reconciliation done" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-k6zwz logSource="pkg/controller/backup_deletion_controller.go:390" |  
time="2023-02-02T06:58:33Z" level=info msg="Removing restores" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-k6zwz logSource="pkg/controller/backup_deletion_controller.go:324" |  
time="2023-02-02T06:58:33Z" level=info msg="Removing backup from backup storage" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-k6zwz logSource="pkg/controller/backup_deletion_controller.go:318" |  
time="2023-02-02T06:58:33Z" level=info msg="Removing pod volume snapshots" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-k6zwz logSource="pkg/controller/backup_deletion_controller.go:310" |  
time="2023-02-02T06:58:33Z" level=info msg="Removing PV snapshots" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-k6zwz logSource="pkg/controller/backup_deletion_controller.go:285" |  
time="2023-02-02T06:58:33Z" level=error msg="Unable to download tarball for backup plausible-clickhouse-lt-20230126053553, skipping associated DeleteItemAction plugins" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-k6zwz error="error copying Backup to temp file: rpc error: code = Unknown desc = error getting object backups/plausible-clickhouse-lt-20230126053553/plausible-clickhouse-lt-20230126053553.tar.gz: NoSuchKey: The specified key does not exist.\n\tstatus code: 404, request id: txd1b409a18f6448dab00e2-0063db5f19, host id: txd1b409a18f6448dab00e2-0063db5f19" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/controller/restore_controller.go:624" error.function=github.com/vmware-tanzu/velero/pkg/controller.downloadToTempFile logSource="pkg/controller/backup_deletion_controller.go:261" |  
time="2023-02-02T06:58:33Z" level=info msg="deletion request 'plausible-clickhouse-lt-20230126053553-qbl89' removed." backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-k6zwz logSource="pkg/controller/backup_deletion_controller.go:444" |  
time="2023-02-02T06:58:33Z" level=info msg="Removing existing deletion requests for backup" backup=plausible-clickhouse-lt-20230126053553 controller=backup-deletion deletebackuprequest=backup/plausible-clickhouse-lt-20230126053553-k6zwz logSource="pkg/controller/backup_deletion_controller.go:427" |  
time="2023-02-02T06:58:33Z" level=info msg="Creating a new deletion request" backup=backup/plausible-clickhouse-lt-20230126053553 expiration="2023-02-02 05:35:53 +0000 UTC" logSource="pkg/controller/gc_controller.go:186"
```

New backups are working as expected. The same error occurs also on other schedules.

**What did you expect to happen:**

The backup should be deleted.

**The following information will help us better understand what's going on**:

[bundle-2023-02-03-09-14-47.tar.gz](https://github.com/vmware-tanzu/velero/files/10576926/bundle-2023-02-03-09-14-47.tar.gz)

**Anything else you would like to add:**

**Environment:**
Latest velero helm release

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by blackpiglet** [comment]
I found the this error message in the backup information.

```
error to open backup repo: repo config /home/nonroot/udmrepo/repo-5263a16b-e3a5-41aa-a51b-f6b318a49255.conf doesn't exist: stat /home/nonroot/udmrepo/repo-5263a16b-e3a5-41aa-a51b-f6b318a49255.conf: no such file or directory
```

This file `/home/nonroot/udmrepo/repo-5263a16b-e3a5-41aa-a51b-f6b318a49255.conf` is used by Velero integrated Kopia library to connect to the Kopia Repository in the OSS.
This file is created and kept in the Velero pod. Still not sure why this file is gone, but it should be recovered after a Velero pod restart.
Could you try to delete the Velero pod, and try whether that will fix your problem?

**Comment by Lyndon-Li** [comment]
One possible scenario:

- The repo connection is established by Velero BackupRepository controller under the circumstance that a new BackupRepository CR is created
- This only needs to be done once, after that, the conf file will cache the connection info in Velero pod
- However, if velero pod restarts, the cache will be lost. At this time, when the related BackupRepository CR still exists, no one will reconnect to the repo

@notz If this is the problem in your env, it could be solved by deleting the related BackupRepository CR. Hopefully you see only one BackupRepository CR in your env, if not, you can try to find the one that match the namespace you are backing up.

**Comment by notz** [comment]
@Lyndon-Li You are right. We redeployed velero via flux/helm 4 days ago. So on restart it looses the files and don't recover them. If i manually delete the BackupRepositiory and trigger a restart, it seems that the files will be created again and the delete requests are succeeding.

**Comment by Lyndon-Li** [comment]
This issue is with the same cause with #5696 and has been fixed by PR #5715 and #5714. 
The fixed has included Velero 1.10.1, please try it out.

**Comment by notz** [comment]
@Lyndon-Li Looks good. All expired backups were deleted on restart.
